### PR TITLE
Add a devicelab task that generates sample app screenshots, etc

### PR DESCRIPTION
**Pull request by HansMuller**
For now this just runs the sample page generator and leaves the generated files and screenshots behind.

### CODE REVIEW

**Comment by HansMuller** [comment]
@yjbanov

File: dev/devicelab/lib/tasks/sample_catalog_generator.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import '../framework/adb.dart';
+import '../framework/framework.dart';
+import '../framework/ios.dart';
+import '../framework/utils.dart';
+
+TaskFunction createSampleCatalogGenerator() {
+  return new SampleCatalogGenerator();
+}
+
+class SampleCatalogGenerator {
```
Comment (yjbanov) [inline dev/devicelab/lib/tasks/sample_catalog_generator.dart]: FYI, this doesn't need to be a class. Can be a function. I other places they are classes for historic reasons.

File: dev/devicelab/manifest.yaml [hunk 2]
```diff
@@ -260,3 +260,19 @@ tasks:
       Measures the performance of Dart VM hot patching feature on Windows.
     stage: devicelab_win
     required_agent_capabilities: ["windows"]
+
+  # Production of sample catalog pages and screenshots
+
+  android_sample_catalog_generator:
```
Comment (yjbanov) [inline dev/devicelab/manifest.yaml]: So far we've been grouping tasks by OS, all "has-android-device" are together, etc. Even though I like this better, let's keep it consistent. I'm ok with either keeping status quo, or regrouping the entire file, if you have extra cycles to do it.

**Review by yjbanov — APPROVED** [review]

**Comment by Hixie** [comment]
The android_sample_catalog_generator task is really flaky.

**Comment by HansMuller** [comment]
The most common swan song for this task is: 
```
Thread(png-cruncher_3): Stream closed
java.io.IOException: Stream closed
	at java.lang.ProcessBuilder$NullOutputStream.write(ProcessBuilder.java:433)
	at java.io.OutputStream.write(OutputStream.java:116)
	at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
	at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
	at sun.nio.cs.StreamEncoder.implFlush(StreamEncoder.java:297)
	at sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:141)
	at java.io.OutputStreamWriter.flush(OutputStreamWriter.java:229)
	at com.android.builder.png.AaptProcess.shutdown(AaptProcess.java:143)
	at com.android.builder.png.QueuedCruncher$1.destruction(QueuedCruncher.java:151)
	at com.android.builder.tasks.WorkQueue.run(WorkQueue.java:245)
	at java.lang.Thread.run(Thread.java:745)
```
I assume that this has something to do with FlutterDriver.screenshot, which the task is using to collect (PNG format) screenshots.  The cruncher code appears to be some version of:

https://github.com/yeungeek/android-platform-tools-base/blob/master/build-system/builder/src/main/java/com/android/builder/png/QueuedCruncher.java#L137

**Pull request merged**